segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -161,14 +163,32 @@ public void setupAppend(SetupAppend setupAppend) {
                 .whenComplete((attributes, u) -> {
                     try {
                         if (u != null) {
+                            if (u instanceof StreamSegmentSealedException) {
```

Comment:
This is an inelegant way of handling this. Please use an `exceptionally` clause before the `whenComplete` to do this.

Better yet, use `Futures.exceptionallyComposeExpecting` with `StreamSegmentSealedException.class` as its expected exception and add your special handling code in the callback.

Reply:
Yes Andrei